src/app/shared/components/admin-carriages/carriage-form/carriage-form.component.ts

```diff
@@ -0,0 +1,124 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { Store } from '@ngrx/store';
+import { Subject, takeUntil } from 'rxjs';
+import { Carriage } from '../../../../features/trips/models/carriage.model';
+import { createCarriage, updateCarriage } from '../../../../core/store/trips/trips.actions';
+
+@Component({
+  selector: 'app-carriage-form',
+  templateUrl: './carriage-form.component.html',
+  styleUrl: './carriage-form.component.scss',
+})
+export class CarriageFormComponent implements OnInit, OnDestroy {
+  @Input() carriage!: Carriage | null;
```

Comment:
I think it's better to write just `@Input() carriage?: Carriage;`

Reply:
Yeah, you're right. I stand corrected.